src/main/java/frc/robot/subsystems/shooter/ShooterIOTalonFX.java

```diff
@@ -0,0 +1,294 @@
+  private StatusSignal<Double> drumMotorStatorCurrentStatusSignal;
+  private StatusSignal<Double> dunkerMotorStatorCurrentStatusSignal;
+  private StatusSignal<Double> angleMotorStatorCurrentStatusSignal;
+
+  // Using StatusSignal to get the supply current of the motors
+  private StatusSignal<Double> shootMotorLeftSupplyCurrentStatusSignal;
+  private StatusSignal<Double> shootMotorRightSupplyCurrentStatusSignal;
+  private StatusSignal<Double> drumMotorSupplyCurrentStatusSignal;
+  private StatusSignal<Double> dunkerMotorSupplyCurrentStatusSignal;
+  private StatusSignal<Double> angleMotorSupplyCurrentStatusSignal;
+
+  // Using StatusSignal to get the velocity of the motors
+  private StatusSignal<Double> shootMotorLeftVelocityStatusSignal;
+  private StatusSignal<Double> shootMotorRightVelocityStatusSignal;
+  private StatusSignal<Double> drumMotorVelocityStatusSignal;
+  private StatusSignal<Double> dunkerMotorVelocityStatusSignal;
+
+  // Creates the reference velocity request for the motors
+  private double shootMotorLeftRequestedVelocityRPS;
+  private double shootMotorRightRequestedVelocityRPS;
+  private double drumMotorRequestedVelocityRPS;
+  private double dunkerMotorVelocityRPS;
+  private double angleMotorRequestedVelocityRPS;
+
+  // Shoot PID Tunable Numbers
+  private final TunableNumber shootMotorsKP = new TunableNumber("Shooter/SHOOT_KP", ShooterConstants.SHOOT_KP);
+  private final TunableNumber shootMotorsKI = new TunableNumber("Shooter/SHOOT_KI", ShooterConstants.SHOOT_KI);
+  private final TunableNumber shootMotorsKD = new TunableNumber("Shooter/SHOOT_KD", ShooterConstants.SHOOT_KD);
+  private final TunableNumber shootMotorsPeakOutput = new TunableNumber("Shooter/SHOOT_PID_PEAK_OUTPUT", ShooterConstants.SHOOT_PID_PEAK_OUTPUT);
+
+  // Angle PID Tunable Numbers
+  private final TunableNumber rotationMotorKP = new TunableNumber("Shooter/ROTATION_KP", ShooterConstants.ROTATION_KP);
+  private final TunableNumber rotationMotorKI = new TunableNumber("Shooter/ROTATION_KI", ShooterConstants.ROTATION_KI);
+  private final TunableNumber rotationMotorKD = new TunableNumber("Shooter/ROTATION_KD", ShooterConstants.ROTATION_KD);
+  private final TunableNumber rotationMotorPeakOutput = new TunableNumber("Shooter/ROTATION_PID_PEAK_OUTPUT", ShooterConstants.ROTATION_PID_PEAK_OUTPUT);
+
+  private TalonFX shootMotorLeft;
+  private TalonFX shootMotorRight;
+  private TalonFX angleMotor;
+  private TalonFX drumMotor;
+  private TalonFX dunkerMotor;
+  private CANcoder angleEncoder;
+  private final DigitalInput shooterSensor;
+
+  public ShooterIOTalonFX() {
+
+    shootMotorLeftVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    shootMotorRightVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    drumMotorVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    dunkerMotorVelocityRequest = new VelocityTorqueCurrentFOC(0);
+
+    shootMotorLeftVelocityStatusSignal = shootMotorLeft.getRotorVelocity();
```

Comment:
I think we should invoke getVelocity which accounts for the sensor to mechanism ratio and, therefore, returns the velocity of the roller, rather than the velocity of the motor. To configure the sensor to mechanism ratio in the config method:

```
config.Feedback.SensorToMechanismRatio = xxx;
```

This is useful because we don't actually care about the velocity of the Falcon shaft. We do care about the velocity of the wheels. So, if we end up changing gear ratios, we only have to change this constant and all the tuning for a specific velocity is still valid.